Shared/Extensions/NSURLExtensions.swift

```diff
@@ -289,6 +289,10 @@ extension URL {
         return host.lowercased() == "localhost" || host == "127.0.0.1"
     }
 
+    public var isValid: Bool {
+        return isWebPage(includeDataURIs: true) && !isLocal
+    }
```

Comment:
I suppose there are cases where `isValid` would false return where `!isLocal` would not. I guess we can keep the extension here, but let's name it something more descriptive (local URLs are still valid URLs). How about `isRemoteWebPage`?